File: cpp/alerts/tracker_announce.cpp
#include "./tracker_announce.h"

namespace ltrs {

uint8_t tracker_announce_alert_get_event(lt::tracker_announce_alert* alert) {
    return static_cast<uint8_t>(alert->event);
}

uint8_t tracker_announce_alert_get_version(lt::tracker_announce_alert* alert) {
    return static_cast<uint8_t>(alert->version);
}

} // namespace ltrs

File: src/ffi/alerts/tracker_announce.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type tracker_announce_alert = crate::ffi::ffi::tracker_announce_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/tracker_announce.h");
        unsafe fn tracker_announce_alert_get_event(alert: *mut tracker_announce_alert) -> u8;
        unsafe fn tracker_announce_alert_get_version(alert: *mut tracker_announce_alert) -> u8;
    }
}
